lib/katello/tasks/upgrades/4.8/fix_incorrect_providers.rake

```diff
@@ -0,0 +1,32 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '4.8' do
+      desc "Fix custom products incorrectly assigned a Red Hat provider"
+      task :fix_incorrect_providers => ["dynflow:client", "katello:check_ping"] do
+        User.current = User.anonymous_admin #set a user for orchestration
+        Dir.glob(Katello::Engine.root.to_s + '/app/models/katello/*.rb').each { |file| require file }
+
+        print "Fixing incorrect providers\n"
+        incorrect_provider_count = 0
+        error_msgs = []
+        Katello::Product.redhat.each do |product|
+          if ::Katello::Glue::Candlepin::Product.custom_product_id?(product.cp_id)
+            print "Fixing provider for #{product.name}\n"
+            incorrect_provider_count += 1
+            product.provider = product.organization.anonymous_provider
+            product.save!
+          end
+        rescue => e
+          print "Error fixing provider: #{e.message}\n"
```

Comment:
This seems like it would be pretty bad & undesirable (an error here).

Should it really be rescued and ignored - I'm actually leaning toward no but that may just be me

Did you actually encounter this case? What was the root cause?

Reply:
nah, was just trying to be safe. I can remove it :+1: